src/main/kotlin/graphics/scenery/Settings.kt

```diff
@@ -9,30 +13,93 @@ import java.util.concurrent.ConcurrentHashMap
             val key = p.key as? String ?: return@forEach
             val value = p.value as? String ?: return@forEach
 
-            if(!key.startsWith("scenery.")) {
+            if(!key.startsWith(prefix)) {
                 return@forEach
             }
 
-            val setting = when {
-                value.lowercase() == "false" || value.lowercase() == "true" -> value.toBoolean()
-                value.lowercase().contains("f") && value.lowercase().replace("f", "").toFloatOrNull() != null -> value.lowercase().replace("f", "").toFloat()
-                value.lowercase().contains("l") && value.lowercase().replace("l", "").toLongOrNull() != null -> value.lowercase().replace("l", "").toLong()
-                value.toIntOrNull() != null -> value.toInt()
-                else -> value
+            val parsed = parseType(value)
+            set(key.substringAfter(prefix), parsed)
+        }
+
+        // Add properties from file, take given file or from params
+        val propertiesFile2 = getOrNull("propertiesFile") ?: propertiesFilePath
+        if (propertiesFile2 != null) {
+            // put properties into settings
+           loadPropertiesFile(File(propertiesFile2))
+        }
+        else {
+            propertiesFile = File(File("").absolutePath + "properties.properties")
+        }
+
+    }
+
+    /**
+     * Loads the .properties [file]
+     * Currently not clearing the old settings -> Overwrites the already set and add new ones. Old stay untouched, if not set by new settings
+     */
+    fun loadPropertiesFile(file : File)
+    {
+        if(file.extension == "properties")
+        {
+            FileInputStream(file).use { input ->
+                val prop = Properties()
+                // load a properties file
+                prop.load(input)
+                prop.propertyNames().toList().forEach { propName ->
+                    set(propName as String, parseType(prop.getProperty(propName)))
+                }
             }
+            propertiesFile = file
+            logger.info("New Properties loaded from $file")
+        }
+        else
+        {
+            logger.warn("${file.absolutePath} is no propertiesFile!")
+        }
+    }
 
-            set(key.substringAfter("scenery."), setting)
+    /**
+     * Saves the currently set settings into [file] if set, or the [propertiesFile] set in [this]
+     */
+    fun saveProperties(file : File? = null)
+    {
+        val props = Properties()
+        this.getAllSettings().sortedDescending().forEach { setting ->
+            props.setProperty(setting, this.getOrNull<String?>(setting).toString())
         }
+        val out : FileOutputStream
+        if(file != null)
+            out = FileOutputStream(file)
+        else
+            out = FileOutputStream(propertiesFile)
+
+        props.store(out, null)
+    }
+
+    /**
+     * Parses the type from the incoming string, returns the casted value
+     */
+    fun parseType(value:String): Any = when {
```

Comment:
I cant be private because I use it in other places too. But I ned to tag it that its used somewhere else (In Microscenery, I need to to parcing there) An alternative would be to move it into a Utilities class, but I think thats not your way to do things with scenery :D